course2/course_en.md

```diff
@@ -0,0 +1,555 @@
+
+```moonbit
+let value = {
+  let x = 1
+  let tmp = x * 2
+  let another_tmp = {
+    let tmp = x * 3
+
+    tmp
+  }
+  tmp + another_tmp + y
+}
+
+let y: Int = 10
+```
+
+---
+
+# Expression Reduction under Variable Binding
+
+- Reduce the expression on the right-hand side of the variable binding.
+- **Replace** occurrences of identifiers with their reduction results.
+- Omit the variable binding part.
+- Reduce the remaining expressions.
+
+```moonbit
+let value = {
+  let x = 1
+  let tmp = 1 * 2 // Replace x
+  let another_tmp = {
+    let tmp = 1 * 3 // Replace x
+
+    tmp
+  }
+  tmp + another_tmp + 10 // Replace y
+}
+
+let y: Int = 10
+```
+
+---
+
+# Expression Reduction under Variable Binding
+
+- Reduce the expression on the right-hand side of the variable binding.
+- **Replace** occurrences of identifiers with their reduction results.
+- Omit the variable binding part.
+- Reduce the remaining expressions.
+
+```moonbit
+let value = {
+  // Omit the variable binding of x
+  let tmp = 2 // Reduce the expression on the right-hand side
+  let another_tmp = {
+    let tmp = 3 // Reduce the expression on the right-hand side
+
+    tmp
+  }
+  tmp + another_tmp + 10
+}
+
+// Omit the variable binding of y
+```
+
+---
+
+# Expression Reduction under Variable Binding
+
+- Reduce the expression on the right-hand side of the variable binding.
+- **Replace** occurrences of identifiers with their reduction results.
+- Omit the variable binding part.
+- Reduce the remaining expressions.
+
+```moonbit
+let value = {
+
+  let tmp = 2
+  let another_tmp = {
+    let tmp = 3
+
+    3 // Replace tmp
+  }
+  tmp + another_tmp + 10
+}
+```
+
+---
+
+# Expression Reduction under Variable Binding
+
+- Reduce the expression on the right-hand side of the variable binding.
+- **Replace** occurrences of identifiers with their reduction results.
+- Omit the variable binding part.
+- Reduce the remaining expressions.
+
+```moonbit
+let value = {
+
+  let tmp = 2
+  let another_tmp = 3 // Reduce the expression on the right-hand side
+  tmp + another_tmp + 10
+}
+```
+
+---
+
+# Expression Reduction under Variable Binding
+
+- Reduce the expression on the right-hand side of the variable binding.
+- **Replace** occurrences of identifiers with their reduction results.
+- Omit the variable binding part.
+- Reduce the remaining expressions.
+
+```moonbit
+let value = {
+
+  let tmp = 2
+  let another_tmp = 3 // Reduce the expression on the right-hand side
+  2 + 3 + 10 // Replace tmp and another_tmp
+}
+```
+
+---
+
+# Expression Reduction under Variable Binding
+
+- Reduce the expression on the right-hand side of the variable binding.
+- **Replace** occurrences of identifiers with their reduction results.
+- Omit the variable binding part.
+- Reduce the remaining expressions.
+
+```moonbit
+let value = 15
+```
+
+---
+
+# Conditional Expression
+
+```
+if condition
+  expression block｜if condition is true
+else
+  expression block｜if condition is false
+```
```

Comment:
I added line breaks where appropriate to achieve a reasonable layout. If translated directly, it cannot be displayed completely in one line.